doc/dynamic-data.md

```diff
@@ -1,4 +1,62 @@
-# Referencing Dynamic Data
+# Referencing dynamic data
 
-Sometimes it can be useful to refer to data in a Container Linux Config that
-isn't known until a machine boots, like it's network address. 
+## Overview
+
+Sometimes it can be useful to refer to data in a Container Linux Config that isn't known until a machine boots, like its network address. This can be accomplished with [coreos-metadata][coreos-metadata] by having it run before a given unit. coreos-metadata will fetch information about the running instance, and write it to a file. The service we want to run can then source this file to get the information when it's starting up.
```

Comment:
This can be accomplished with coreos-metadata. coreos-metadata is a very basic utility that fetches information about the current machine and makes it available for consumption. By making it a dependency of services which requires this information, systemd will ensure that coreos-metadata has successfully completed before starting these services. These services can then simply source the fetched information and let systemd perform the environment variable expansions.